nixos/modules/image/repart-image.nix

```diff
@@ -0,0 +1,69 @@
+{ name, fileSystems, partitions, split, seed, definitionsDirectory }:
+
+{ lib
+, runCommand
+, python3
+, black
+, ruff
+, mypy
+, systemd
+, fakeroot
+, util-linux
+, dosfstools
+, mtools
+, e2fsprogs
+, squashfsTools
+, erofs-utils
+, btrfs-progs
+, xfsprogs
+}:
+
+let
+  amendRepartDefinitions = runCommand "amend-repart-definitions.py"
+    {
+      nativeBuildInputs = [ python3 black ruff mypy ];
+    } ''
+    install ${./amend-repart-definitions.py} $out
+    patchShebangs --build $out
+
+    black --check --diff $out
```

Comment:
You can use `ruff format` here which is a drop in replacement for black and faster.

Reply:
```suggestion
    ruff format --check --diff $out
```

Reply:
Just saw that you were just moving out this expression and this was already in use before. Feel free to skip this change or do this in another PR.

Reply:
I'll defer this to another PR as well.